monai/transforms/compose.py

```diff
@@ -358,3 +359,61 @@ def inverse(self, data):
         for o in reversed(applied_order):
             data = apply_transform(self.transforms[o].inverse, data, self.map_items, self.unpack_items, self.log_stats)
         return data
+
+
+class SomeOf(Compose):
+    """
+    ``SomeOf`` provides the ability to randomly sample a fixed number or
+    up to a fixed number of transforms from a list of callables.
+
+    If sampling up to the max_samples, each invocation of SomeOf uniformly samples a target number of transforms
+    for that invocation.
+
+    At each invocation the set of transforms to apply is uniformly sampled without replacement.
+
+    Args:
+        transforms: list of callables.
+        map_items: whether to apply transform to each item in the input `data` if `data` is a list or tuple.
+            Defaults to `True`.
+        unpack_items: whether to unpack input `data` with `*` as parameters for the callable function of transform.
+            Defaults to `False`.
+        log_stats: whether to log the detailed information of data and applied transform when error happened,
+            for NumPy array and PyTorch Tensor, log the data shape and value range,
+            for other metadata, log the values directly. Default to `False`.
+        min_samples: minimum number of transforms to sample. Defaults to `0`.
+        max_samples: maximum number of transforms to sample. Defaults to `3`.
+        fixed: whether to sample exactly `max_samples` transforms, or up to it. Defaults to `False`.
+    """
+
+    def __init__(
+        self,
+        transforms: Sequence[Callable] | Callable | None = None,
+        map_items: bool = True,
+        unpack_items: bool = False,
+        log_stats: bool = False,
+        min_samples: int = 0,
+        max_samples: int = 3,
+        fixed: bool = True,
+    ) -> None:
+        super().__init__(transforms, map_items, unpack_items, log_stats)
+        if transforms is None:
+            transforms = []
+        self.original_transforms = transforms
+        self.n_transforms = len(transforms)
+        self.max_samples = min(self.n_transforms, max_samples)
+        self.fixed = fixed
+        self.min_samples = min_samples
+        self.random = random.Random(get_seed())
```

Comment:
I would initialise `self.active` here so that it always exists rather than popping in when `__call__` is called for the first time. It's initial value can just be `list(range(len(transforms)))`.

Reply:
No longer needed after switching to self.R from Randomizable class.  self.transforms is set in the call to `super().__init__` which has the effect of what you're suggesting.